src/indexer/redeem_fees.rs

```diff
@@ -34,22 +45,93 @@ impl Indexer {
+            let wallet = self.get_or_create_wallet(&fee.mint).await?;
+            info!("redeeming into {}", wallet.id);
         }
 
         Ok(())
     }
 
     /// Find or create a wallet to store balances of a given mint
-    fn get_or_create_wallet(&mut self, mint: &str) -> Result<(), String> {
+    async fn get_or_create_wallet(&mut self, mint: &str) -> Result<WalletMetadata, String> {
         let maybe_wallet = self.get_wallet_for_mint(mint)?;
         let maybe_wallet =
             maybe_wallet.or_else(|| self.find_wallet_with_empty_balance().ok().flatten());
 
-        if maybe_wallet.is_none() {
-            println!("creating new wallet for {mint}");
+        match maybe_wallet {
+            Some(wallet) => Ok(wallet),
+            None => {
+                info!("creating new wallet for {mint}");
+                self.create_new_wallet().await
+            }
         }
+    }
 
-        Ok(())
+    /// Create a new wallet for managing a given mint
+    ///
+    /// Return the new wallet's metadata
+    async fn create_new_wallet(&mut self) -> Result<WalletMetadata, String> {
+        // 1. Create the new wallet on-chain
+        let (wallet_id, root_key) = self.create_renegade_wallet().await?;
+
+        // 2. Create a secrets manager entry for the new wallet
+        let secret_name = self
+            .create_secrets_manager_entry(wallet_id, root_key)
+            .await?;
+
+        // 3. Add an entry in the wallets table for the newly created wallet
+        let entry = WalletMetadata::empty(wallet_id, secret_name);
+        self.insert_wallet(entry.clone())?;
+
+        Ok(entry)
+    }
+
+    /// Create a new Renegade wallet on-chain
+    async fn create_renegade_wallet(&mut self) -> Result<(WalletIdentifier, LocalWallet), String> {
+        let chain_id = self
+            .client
+            .chain_id()
+            .await
+            .map_err(raw_err_str!("Error fetching chain ID: {}"))?;
+        let root_key = LocalWallet::new(&mut thread_rng());
+
+        let wallet_id = derive_wallet_id(&root_key)?;
+        let blinder_seed = derive_blinder_seed(&root_key)?;
+        let share_seed = derive_share_seed(&root_key)?;
+        let key_chain = derive_wallet_keychain(&root_key, chain_id)?;
+
+        let wallet = Wallet::new_empty_wallet(wallet_id, blinder_seed, share_seed, key_chain);
+        create_new_wallet(wallet, &self.relayer_url).await?;
+        info!("created new wallet for fee redemption");
+
+        Ok((wallet_id, root_key))
+    }
+
+    /// Add a Renegade wallet to the secrets manager entry so that it may be recovered later
+    ///
+    /// Returns the name of the secret
+    async fn create_secrets_manager_entry(
+        &mut self,
+        id: WalletIdentifier,
+        wallet: LocalWallet,
+    ) -> Result<String, String> {
+        let client = SecretsManagerClient::new(&self.aws_config);
+        let secret_name = format!("redemption-wallet-{id}");
```

Comment:
nit: this would bubble up through the function interfaces, but it'd be nice to add the env to this name

Reply:
done